core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -193,12 +194,11 @@ private[spark] class DiskBlockObjectWriter(
     }
 
     objOut.writeObject(value)
+    numRecordsWritten += 1
```

Comment:
The current use of `numRecordsWritten` assumes that a `BlockObjectWriter` will never be closed and re-opened, correct? I'm not sure whether that is enforced anywhere. Should we zero this var out in `open()`?